diff --git a/answers.module b/answers.module
index 061f0bf..1573d12 100644
--- a/answers.module
+++ b/answers.module
@@ -104,7 +104,7 @@ function answers_node_view($node, $view_mode, $langcode) {
   }
   elseif (($node->type == 'answer') &&
           variable_get('answers_redirect_from_answer_to_question_nodes_p', TRUE) &&
-          ($view_mode == 'full') && 
+          (($view_mode == 'full') || ($view_mode == 'page_manager')) &&
           (node_is_page($node)) &&
           ($node->status)) {
     // If viewing the node page for a published answer and if configured to use theme templates,
