diff --git a/modules/file/file.field.inc b/modules/file/file.field.inc
index c841f60..a992119 100644
--- a/modules/file/file.field.inc
+++ b/modules/file/file.field.inc
@@ -216,10 +216,11 @@ function file_field_presave($entity_type, $entity, $field, $instance, $langcode,
   // permanent status, so that it will not be removed when temporary files are
   // cleaned up.
   foreach ($items as $item) {
-    $file = file_load($item['fid']);
-    if (!$file->status) {
-      $file->status = FILE_STATUS_PERMANENT;
-      file_save($file);
+    if ($item && $file = file_load($item['fid'])) {
+      if (!$file->status) {
+        $file->status = FILE_STATUS_PERMANENT;
+        file_save($file);
+      }
     }
   }
 }
