diff --git a/flickr/flickr.inc b/flickr/flickr.inc
index 2d59f7d..89bbe28 100644
--- a/flickr/flickr.inc
+++ b/flickr/flickr.inc
@@ -135,7 +135,7 @@ function flickr_request($method, $args, $cacheable = TRUE, $return_errors = FALS
 
   // Make sure it unserializes.
 
-  $response = unserialize($result->data);
+  $response = @unserialize($result->data);
   if (!$response) {
     if ($return_errors) {
       return array(
