diff --git a/flickr/flickr.inc b/flickr/flickr.inc
index 2d59f7d..de41a38 100644
--- a/flickr/flickr.inc
+++ b/flickr/flickr.inc
@@ -103,6 +103,11 @@ function flickr_request($method, $args, $cacheable = TRUE, $return_errors = FALS
   foreach ($args as $k => $v) {
     $encoded_params[] = urlencode($k) . '=' . urlencode($v);
   }
+  
+  if (!in_array('format=php_serial', $encoded_params)) {
+    $encoded_params[1] = 'format=php_serial';
+  }
+  
   $url = FLICKR_REST_ENDPOINT . '?' . implode('&', $encoded_params);
 
   // If it's a cachable request, try to load a cached value.
