diff --git a/panels_ipe/plugins/display_renderers/panels_renderer_ipe.class.php b/panels_ipe/plugins/display_renderers/panels_renderer_ipe.class.php
index f1b32c5..e500b20 100644
--- a/panels_ipe/plugins/display_renderers/panels_renderer_ipe.class.php
+++ b/panels_ipe/plugins/display_renderers/panels_renderer_ipe.class.php
@@ -153,7 +153,7 @@ class panels_renderer_ipe extends panels_renderer_editor {
 
       // Get the administrative title.
       $content_type = ctools_get_content_type($pane->type);
-      $title = ctools_content_admin_title($content_type, $pane->subtype, $pane->configuration, $this->display->context);
+      $title = ctools_content_get_subtype($content_type, $pane->subtype) ? ctools_content_admin_title($content_type, $pane->subtype, $pane->configuration, $this->display->context) : t('Unknown');
 
       $content->content = t('Placeholder for empty or inaccessible "@title"', array('@title' => html_entity_decode($title, ENT_QUOTES)));
       // Add these to prevent notices.
